### PR TITLE
Enforce tsdocs for exported functions

### PR DESCRIPTION
**Pull request by Brainicism**

### CODE REVIEW

File: src/commands/game_commands/exp.ts [hunk 1]
```diff
@@ -58,6 +58,12 @@ interface ExpModifier {
     isPenalty: boolean;
 }
 
+/**
+ * Calculates the exp multiplier based on the round options
+ * @param guildPreference - The guild preference
+ * @param voteBonusExp - Whetther bonus EXP should be applied to the modifier
+ * @returns an array describing the EXP modifiers activated and their numierical value
```
Comment (taahamahdi) [inline src/commands/game_commands/exp.ts:65]: numerical

File: src/commands/game_commands/exp.ts [hunk 2]
```diff
@@ -58,6 +58,12 @@ interface ExpModifier {
     isPenalty: boolean;
 }
 
+/**
+ * Calculates the exp multiplier based on the round options
+ * @param guildPreference - The guild preference
+ * @param voteBonusExp - Whetther bonus EXP should be applied to the modifier
```
Comment (taahamahdi) [inline src/commands/game_commands/exp.ts:64]: Whether

File: src/commands/game_commands/exp.ts [hunk 3]
```diff
@@ -135,10 +141,22 @@ async function calculateOptionsExpMultiplier(guildPreference: GuildPreference, v
     return (await calculateOptionsExpMultiplierInternal(guildPreference, voteBonusExp)).reduce((a, b) => ExpBonusModifierValues[b.name] * a, 1);
 }
 
+/**
+ * @param numParticipants - The number of participants
+ * @returns the EXP modifiered based onthe number of participants
```
Comment (taahamahdi) [inline src/commands/game_commands/exp.ts:146]: Modifier

On the

File: src/commands/game_commands/exp.ts [hunk 4]
```diff
@@ -168,6 +186,16 @@ export function calculateRoundExpMultiplier(gameRound: GameRound, numParticipant
...
 }
 
+/**
+ * @param guildPreference - The guild preference
+ * @param gameRound - The game round
+ * @param numParticipants - The number of participants
+ * @param streak - The current guessing streak
+ * @param guessSpeed - The guess speed
+ * @param place - The place of the guess
+ * @param voteBonusExp - Whetther bonus EXP should be applied to the modifier
```
Comment (taahamahdi) [inline src/commands/game_commands/exp.ts:196]: Whether

File: src/commands/game_commands/hint.ts [hunk 5]
```diff
@@ -45,6 +45,13 @@ async function sendHintNotification(message: GuildTextableMessage, gameSession:
     }
 }
 
+/**
+ * @param gameSession - The game session
+ * @param guildPreference - The guild preference
+ * @param gameRound - The game round
+ * @param message - The originiating messages
```
Comment (taahamahdi) [inline src/commands/game_commands/hint.ts:52]: Originating message

File: src/events/client/shardDisconnect.ts [hunk 6]
```diff
@@ -2,6 +2,11 @@ import { IPCLogger } from "../../logger";
 
 const logger = new IPCLogger("shardDisconnect");
 
+/**
+ * Handles the 'shardDisconnect' event.
```
Comment (taahamahdi) [inline src/events/client/shardDisconnect.ts:6]: Other comments don't end with periods

File: src/events/client/voiceChannelSwitch.ts [hunk 7]
```diff
@@ -2,6 +2,12 @@ import Eris from "eris";
 import { state } from "../../kmq_worker";
 import { checkBotIsAlone } from "../../helpers/discord_utils";
 
+/**
+ * Handles teh 'voiceChannelSwitch' event
```
Comment (taahamahdi) [inline src/events/client/voiceChannelSwitch.ts:6]: The

File: src/helpers/discord_utils.ts [hunk 8]
```diff
@@ -374,7 +377,11 @@ export async function sendInfoMessage(messageContext: MessageContext, embedPaylo
  * @param messageContext - An object to pass along relevant parts of Eris.Message
  * @param scoreboard - The GameSession's corresponding Scoreboard
  * @param gameRound - The GameSession's corresponding GameRound
- * @param songGuessed - Whether the song was guessed
+ * @param guessModeType - The type of guess mode
+ * @param playerRoundResults - The player round results
+ * @param isMultipleChoiceMode  - Whether the game is in multiple choice mode
+ * @param timeRemaining - The time remaining for the timer option
```
Comment (taahamahdi) [inline src/helpers/discord_utils.ts:383]: For the duration option

File: src/scripts/backup-kmq-database.ts [hunk 9]
```diff
@@ -16,6 +16,9 @@ program
 program.parse();
 const options = program.opts();
 
+/**
+ *
```
Comment (taahamahdi) [inline src/scripts/backup-kmq-database.ts:20]: Missing

File: src/scripts/download-new-songs.ts [hunk 10]
```diff
@@ -10,7 +10,8 @@ import { retryJob, getAudioDurationInSeconds } from "../helpers/utils";
 
 const logger = new IPCLogger("download-new-songs");
 const TARGET_AVERAGE_VOLUME = -30;
-export async function clearPartiallyCachedSongs(): Promise<void> {
+
+async function clearPartiallyCachedSongs(): Promise<void> {
```
Comment (taahamahdi) [inline src/scripts/download-new-songs.ts:14]: Missing doc

File: src/structures/guild_preference.ts [hunk 11]
```diff
@@ -118,8 +118,9 @@ export const GameOptionInternalToGameOption: { [option: string]: string } = {
 };
 
 /**
- * @param text - The text to truncate
- * @param length - The number of characters to truncate to
+ * @param groups - The artists to combine into a string
+ * @param truncate - Whether to truncuate the final result
+ * @param spaceDelimiter - Wheter to space delimit the final result
```
Comment (taahamahdi) [inline src/structures/guild_preference.ts:123]: Whether

File: src/structures/team_scoreboard.ts [hunk 12]
```diff
@@ -171,8 +172,8 @@ export default class TeamScoreboard extends Scoreboard {
     }
 
     /**
-     * @returns the name of the player associated with the given userID
-     *  @returns the player's tag
+     * @param userID - The unique identifier of the player whose score is being incremented
```
Comment (taahamahdi) [inline src/structures/team_scoreboard.ts:175]: The player's score is not being incremented

**Review by taahamahdi — CHANGES_REQUESTED** [review]

File: src/scripts/download-new-songs.ts (hunk 10 above)
Comment (Brainicism) [reply src/scripts/download-new-songs.ts:14]: Function isn't exported

**Review by taahamahdi — APPROVED** [review]

**Pull request merged**